generic-worker: panic "close of closed channel" in Command.Kill()

## Summary

`Command.Kill()` can panic with "close of closed channel" when called more than once on the same command. This affects all platforms (Windows, Linux, macOS, FreeBSD).

## Observed behavior

On generic-worker 96.2.3 on Windows, we got this panic during a graceful-termination:

```
panic: close of closed channel

goroutine 5 [running]:
github.com/taskcluster/taskcluster/v96/workers/generic-worker/process.(*Command).Kill(0xc00042c340)
  workers/generic-worker/process/multiuser_windows.go:137 +0x54
main.(*TaskRun).kill(0xc000500808)
  workers/generic-worker/main.go:896 +0x77
main.(*TaskStatusManager).Abort.func1(0xc000500808)
  workers/generic-worker/taskstatus.go:211 +0x4f
main.(*TaskStatusManager).updateStatus(...)
  workers/generic-worker/taskstatus.go:290 +0x1b8
main.(*TaskStatusManager).Abort(...)
  workers/generic-worker/taskstatus.go:207 +0x91
main.(*AbortTaskFeature).Start.func1(0x75?)
  workers/generic-worker/abort_feature.go:61 +0xba
github.com/taskcluster/taskcluster/v96/workers/generic-worker/graceful.Terminate(0x90?)
  workers/generic-worker/graceful/graceful.go:54 +0x91
main.startProtocol.func1(...)
  workers/generic-worker/workerrunner.go:86 +0xde
github.com/taskcluster/taskcluster/v96/tools/workerproto.(*Protocol).recvLoop(...)
  tools/workerproto/protocol.go:154 +0x18d
```

## Root cause

`Command.Kill()` unconditionally calls `close(c.abort)` with no guard against double-close:

- **`process/posix.go:61`** — affects linux, darwin, freebsd
- **`process/multiuser_windows.go:137`** — affects windows

Multiple code paths can call `task.kill()` (which calls `command.Kill()` for each command):

1. **Reclaim failure** (`taskstatus.go:172`) — kills the task when a reclaim fails
2. **Abort** (`taskstatus.go:214`) — kills the task on abort (graceful termination, max run time, resource limits)

These run in different goroutines. The `TaskStatusManager` lock prevents some races (e.g. two concurrent `Abort` calls), but the reclaim-failure path calls `task.kill()` inside the `reclaimed` status handler — a different status transition than `Abort` — so the lock doesn't protect against a reclaim failure followed by an abort.

## Proposed fix

Use `sync.Once` to guard the channel close in `Kill()`:

```go
// Add to Command struct: abortOnce sync.Once

func (c *Command) Kill() (killOutput string, err error) {
    c.abortOnce.Do(func() { close(c.abort) })
    // ... rest unchanged
}
```

This needs to be applied in both `posix.go` and `multiuser_windows.go`, and the `Command` struct definitions need the new `abortOnce sync.Once` field.

--- a/changelog/issue-8410.md
+++ b/changelog/issue-8410.md
@@ -1,0 +1,5 @@
+audience: worker-deployers
+level: patch
+reference: issue 8410
+---
+Generic Worker: Fix panic "close of closed channel" in `Command.Kill()` when multiple abort paths (e.g., reclaim failure and graceful termination) race to kill a task's processes.

--- a/workers/generic-worker/process/multiuser_darwin.go
+++ b/workers/generic-worker/process/multiuser_darwin.go
@@ -37,7 +37,8 @@ type (
 		// abort channel is closed when Kill() is called so that Execute() can
 		// return even if cmd.Wait() is blocked. This is useful since cmd.Wait()
 		// sometimes does not return promptly.
-		abort chan struct{}
+		abort     chan struct{}
+		abortOnce sync.Once
 		// Once command has run, Result is updated (similar to cmd.ProcessState)
 		result *Result
 		conn   net.Conn

--- a/workers/generic-worker/process/multiuser_windows.go
+++ b/workers/generic-worker/process/multiuser_windows.go
@@ -134,7 +134,7 @@ func NewCommandNoOutputStreams(commandLine []string, workingDirectory string, en
 
 func (c *Command) Kill() (killOutput string, err error) {
 	// abort even if process hasn't started
-	close(c.abort)
+	c.abortOnce.Do(func() { close(c.abort) })
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
 	if c.Process == nil {

--- a/workers/generic-worker/process/other.go
+++ b/workers/generic-worker/process/other.go
@@ -19,7 +19,8 @@ type (
 		// abort channel is closed when Kill() is called so that Execute() can
 		// return even if cmd.Wait() is blocked. This is useful since cmd.Wait()
 		// sometimes does not return promptly.
-		abort chan struct{}
+		abort     chan struct{}
+		abortOnce sync.Once
 		// Once command has run, Result is updated (similar to cmd.ProcessState)
 		result *Result
 	}

--- a/workers/generic-worker/process/posix.go
+++ b/workers/generic-worker/process/posix.go
@@ -58,7 +58,7 @@ func (c *Command) SetEnv(envVar, value string) {
 
 func (c *Command) Kill() (killOutput string, err error) {
 	// abort even if process hasn't started
-	close(c.abort)
+	c.abortOnce.Do(func() { close(c.abort) })
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
 	// if pid has been set in result, use that

